R/db.R

```diff
@@ -9,9 +13,22 @@ db_connect <- function(type, path) {
   x <- dettl_db_args(path, type)
   con <- do.call(DBI::dbConnect,
                  c(list(x$driver()), x$args))
+  sqlite_enable_fk(con)
   con
 }
 
+#' Enable foreign key constraints for SQLite connections
+#'
+#' Foreign key constraints aren't enabled by default in SQLite. Ensure
+#' they are enabled each time we connect to the db.
+#'
+#' @keywords internal
+sqlite_enable_fk <- function(con) {
+  if (class(con) == "SQLiteConnection") {
```

Comment:
Can you please update this to be `inherits(con, "SQLiteConnection")` - I should have picked that up before but that's the only safe way to ask about class membership (class vectors can be any length which makes `==` work element-wise)